src/CVNBot/ProjectList.cs

```diff
@@ -56,7 +56,8 @@ public void LoadFromFile()
         /// </summary>
         /// <param name="projectName">Name of the project (e.g., en.wikipedia) to add</param>
         /// <param name="interwiki">Interwiki link (e.g., it:s: -- can be empty string)</param>
-        public void AddNewProject(string projectName, string interwiki)
+        /// <param name="lang">ISO 639 lang code (e.g. "it", "es" Optional) </param>
+        public void AddNewProject(string projectName, string interwiki, string lang = "")
```

Comment:
This appears to be a separate feature, and not yet complete. I like the idea, but if I understand correctly, it is not yet complete, and might be easier to review as a separate pull request (or at least a separate commit in the same pull request).

Reply:
This adding argument is can not separate the patch. because some specific projects are can not get language information from URL. (e.g. nyc.wikimedia or future non language code *.wikimedia sites)

Also `lang` arg is passed to Project instance in L130

Reply:
Aha, I see now. The language code is needed for `amlang` parameter.

That parameter is not required, however. If you omit it, the default language is the content language. Which the software internally has configured. We do not need to know it inside CVNBot.

That will be easier I think, right?

When logged-in, the API uses your user preferences as default. But when logged-out, it uses the default/content language. Maybe that is why you thought it was needed?

If we want to have a url that always works, we can use `uselang=content`. Which will make it the same even if you test it when you are logged-in with a different language preference.

For example, <https://nl.wikimedia.org/w/api.php?action=query&meta=allmessages&ammessages=3d-desc&uselang=content>.

Reply:
Thanks for pointing it. Yeah, you're right.